Assembus/Assets/Scripts/Shared/DataExport.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.IO;
+using System.Xml.Linq;
+using Models.Project;
+using Services.Serialization;
+using UnityEngine;
+
+namespace Shared
+{
+    public static class DataExport
+    {
+        private static readonly ProjectManager ProjectManager = ProjectManager.Instance;
+
+        /// <summary>
+        ///     Export assembly data to a given file
+        /// </summary>
+        /// <param name="exportPath">Path to save the xml file to</param>
+        /// <returns>A boolean indicating the success and an error message in case of failure</returns>
+        public static (bool, string) ExportData(string exportPath)
+        {
+            // Save chosen file name
+            ProjectManager.CurrentProject.ExportFileName = Path.GetFileName(exportPath);
+
+            try
+            {
+                // Export data structure to XML file
+                ConvertToExportXml().Save(exportPath);
+            }
+            catch (ModelManager.ToplevelComponentException e)
+            {
+                return (false, e.ComponentName + " can't be on top level!");
+            }
+            catch (IOException)
+            {
+                return (false, "Error while exporting data!");
+            }
+            return (true, "");
+        }
+
+        /// <summary>
+        ///     Convert the model data to the export XML format
+        /// </summary>
+        /// <returns>XElement representing the current internal data structure</returns>
+        private static XElement ConvertToExportXml()
```

Comment:
The filename of the used OBJ-file is not saved in the exported XML file. Is this the intended behaviour?